kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiDefaultStream.java

```diff
@@ -0,0 +1,278 @@
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * An extension of {@link StorageWriteApiBase} which uses default streams to write data following at least once semantic
+ */
+public class StorageWriteApiDefaultStream extends StorageWriteApiBase {
+    private static final Logger logger = LoggerFactory.getLogger(StorageWriteApiDefaultStream.class);
+    private static final ConcurrentMap<String, JsonStreamWriter> tableToStream = new ConcurrentHashMap<>();
+
+    public StorageWriteApiDefaultStream(int retry,
+                                        long retryWait,
+                                        BigQueryWriteSettings writeSettings,
+                                        boolean autoCreateTables,
+                                        ErrantRecordHandler errantRecordHandler,
+                                        SchemaManager schemaManager) {
+        super(retry, retryWait, writeSettings, autoCreateTables, errantRecordHandler, schemaManager);
+    }
+
+    @Override
+    public void shutdown() {
+        logger.info("Closing all writer for default stream on all tables");
+        tableToStream.keySet().forEach(this::closeAndDelete);
+        logger.info("Closed all writer for default stream on all tables");
+    }
+
+    /**
+     * Either gets called when shutting down the task or when we receive exception that the stream
+     * is actually closed on Google side. This will close and remove the stream from our cache.
+     *
+     * @param tableName The table name for which stream has to be removed.
+     */
+    private void closeAndDelete(String tableName) {
+        logger.debug("Closing stream on table {}", tableName);
+        if (tableToStream.containsKey(tableName)) {
+            synchronized (tableToStream) {
+                tableToStream.get(tableName).close();
+                tableToStream.remove(tableName);
+            }
+            logger.debug("Closed stream on table {}", tableName);
+        }
+    }
+
+    /**
+     * Open a default stream on table if not already present
+     *
+     * @param table The table on which stream has to be opened
+     * @param rows  The input rows (would be sent while table creation to identify schema)
+     * @return JSONStreamWriter which would be used to write data to bigquery table
+     */
+    @VisibleForTesting
+    public JsonStreamWriter getDefaultStream(TableName table, List<Object[]> rows) {
+        String tableName = table.toString();
+        return tableToStream.computeIfAbsent(tableName, t -> {
+            boolean tableCreationOrUpdateAttempted = false;
+            int additionalRetries = 0;
+            int additionalWait = 0;
+            Exception mostRecentException;
+            int attempt = 0;
+            do {
+                try {
+                    if (attempt > 0) {
+                        waitRandomTime(additionalWait);
+                    }
+                    return JsonStreamWriter.newBuilder(t, getWriteClient()).build();
```

Comment:
we should allow other parameters of JsonStreamWriter to configurable e.g.
```
ignoreUnknownValues
customExecutorProvider
enableConnectionMultiplexing
```

Reply:
Thanks for the inputs @SubhamSinghal . Currently connector does not expose these configs. Making JsonStreamWriter configurable would mean adding adding these configs as user-configurable that is currently not in scope for first iteration. Though we can look into this and prioritise in future iterations.